packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -228,6 +230,92 @@ export const saveDataset = async (
 	return createDatasetFromSimulationResult(projectId, simulationId, datasetName);
 };
 
+/**
+ * This is a client side function to create a CsvAsset similar to the getCsv function from DatasetResource.java
+ * The reason we reimplement this client side is because runResults already has all of the data we need, except
+ * it's not in the correct format necessary to be consumed by the TeraDatasetDatatable component.
+ * @param runResults
+ * @param runId
+ * @returns CsvAsset object with the data from the runResults
+ */
+const createCsvAssetFromRunResults = (runResults: RunResults, runId?: string): CsvAsset | null => {
+	const runResult: RunResults = _.cloneDeep(runResults);
```

Comment:
just `import { cloneDeep } from lodash;`
```suggestion
	const runResult: RunResults = cloneDeep(runResults);
```